packages/cache/src/internal/cacheHttpClient.ts

```diff
@@ -256,6 +260,64 @@ export async function downloadCache(
   }
 }
 
+async function downloadCacheStorageSDK(
+  archiveLocation: string,
+  archivePath: string
+): Promise<void> {
+  const client = new BlockBlobClient(archiveLocation)
+  const properties = await client.getProperties()
+  const contentLength = properties.contentLength ?? -1
+  
+  if (contentLength < 0) {
+    // We should never hit this condition, but just in case fall back to downloading the
+    // file as one large stream.
+    core.debug('Unable to determine content length, downloading file as stream...')
+    await client.downloadToFile(archivePath)
+  } else {
+    // Use downloadToBuffer for faster downloads, since internally it splits the
+    // file into 4 MB chunks which can then be parallelized and retried independently.
+    // Note that the max buffer length is 1 GB on 32-bit systems and 2 GBs on 64-bit
+    // systems. If the content length exceeds this limit, split the download into
+    // multiple calls.
+    const maxSegmentSize = buffer.kMaxLength
+    let offset = 0
+
+    const fd = fs.openSync(archivePath, 'w')
+
+    while (offset < contentLength) {
+      const segmentSize = Math.min(maxSegmentSize, contentLength-offset)
+      core.debug(`Downloading segment at offset ${offset} with length ${segmentSize}...`)
+      
+      const result = await client.downloadToBuffer(offset, segmentSize, { concurrency: os.cpus().length * 8 })
+      fs.writeFileSync(fd, result)
+
+      core.debug(`Finished segment at offset ${offset}`)
+      offset += segmentSize
+    }
+
+    fs.closeSync(fd)
```

Comment:
Should we need to put this in a `try / finally` block in case `downloadToBuffer` throws an error? We want to ensure we close the `fd` regardless of any errors

Reply:
Fixed, added try-finally block